dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -37,148 +43,160 @@
-						bundleStream.close();
-						Logger.error(this.getClass(), "Push Publishing failed from " + remoteIP + " invalid endpoint or token");
-						
-		        return responseResource.responseError( HttpStatus.SC_UNAUTHORIZED );
-		       }
-
-					String bundlePath = ConfigUtils.getBundlePath()+File.separator+MY_TEMP;
-					String bundleFolder = fileName.substring(0, fileName.indexOf(".tar.gz"));
-
-		            PublishAuditStatus status = PublishAuditAPI.getInstance().updateAuditTable( mySelf.getId(), mySelf.getId(), bundleFolder, true );
-
-		            if(bundleName.trim().length()>0) {
-					    // save bundle if it doesn't exists
-		                Bundle foundBundle = APILocator.getBundleAPI().getBundleById( bundleFolder );
-		                if ( foundBundle == null || foundBundle.getId() == null ) {
-		                    Bundle bundle = new Bundle();
-							bundle.setId(bundleFolder);
-							bundle.setName(bundleName);
-							bundle.setPublishDate(Calendar.getInstance().getTime());
-							bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
-							bundle.setForcePush(forcePush);
-		                    APILocator.getBundleAPI().saveBundle(bundle);
-					    }
-					}
-
-					//Write file on FS
-					FileUtil.writeToFile(bundleStream, bundlePath+fileName);
-
-					//Start thread
-					if(!status.getStatus().equals(Status.PUBLISHING_BUNDLE)) {
-						new Thread(new PublishThread(fileName, groupId, endpointId, status)).start();
-					}
-
-					HibernateUtil.commitTransaction();
-
-					return Response.status(HttpStatus.SC_OK).build();
-				} catch (NumberFormatException e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				} catch (Exception e) {
-				    try {
-		                HibernateUtil.rollbackTransaction();
-		            } catch (DotHibernateException e1) {
-		                Logger.error(this, "error rollback",e1);
-		            }
-					Logger.error(PublisherQueueJob.class, "Error caused by remote call of: "+remoteIP);
-					Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-				}
-				finally {
-				    try {
-		                HibernateUtil.closeSession();
-		            } catch (DotHibernateException e) {
-		                Logger.error(this, "error close session",e);
-		            }
-				}
-    		}
-    	} catch (IOException e) {
-    		Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-    	}
-
-		return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+			@QueryParam("type")        final String type,
+			@QueryParam("callback")    final String callback,
+			@QueryParam("FORCE_PUSH")  final boolean forcePush,
+			@Context final HttpServletRequest  request,
+			@Context final HttpServletResponse response
+	) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)